build-scripts/build_course_zips.js

```diff
@@ -18,65 +18,48 @@ const newProgressBar = () => {
   )
 }
 
-const options = yargs
-  .usage("Usage: -d <dist> -c <courses> -z <zips>")
-  .option("d", {
-    alias:        "dist",
-    describe:     "distribution path",
-    type:         "string",
-    demandOption: true
-  })
-  .option("c", {
-    alias:        "courses",
-    describe:     "course markdown path",
-    type:         "string",
-    demandOption: true
-  })
-  .option("z", {
-    alias:        "zips",
-    describe:     "zip output path",
-    type:         "string",
-    demandOption: false
-  }).argv
-
-const distPath = options.dist
-const coursesPath = options.courses
-const zipsPath = options.zips || "zips"
-
 // clear out the distribution path and run the webpack build
-const run = async () => {
+const buildZips = async (
+  distPath = "dist",
+  coursesPath = "site/content/courses",
+  zipsPath = "zips",
+  staticAssetsPath,
+  staticPrefix
+) => {
   if ((await execFile("which", ["zip"])).error) {
     throw new Error("Unable to find zip binary")
   }
 
-  const baseDir = path.resolve(coursesPath, "..", "..")
+  // remove any preceeding slash on the staticPrefix, since it will be used as a path within the zip
+  if (staticPrefix) {
+    staticPrefix = staticPrefix.startsWith("/") ?
+      staticPrefix.substring(1) :
+      staticPrefix
+  }
+
+  const baseDir = process.cwd()
   const relative = path.relative(baseDir, zipsPath)
-  if (relative && !relative.startsWith("..") && !path.isAbsolute(relative)) {
+  if (
+    relative &&
+    (relative.startsWith(path.join("site", "content")) ||
+      relative.startsWith(path.join("site", "static")))
+  ) {
     // make sure zips are not picked up by hugo, causing a blowup in file size
```

Comment:
There is no `coursesPath` in the logic above so how would we know this now?

Reply:
This script is only ever run through the package script `build:zips`, which is run in the project directory with `npm run build:zips`.  That ensures that `process.cwd()` will always return the project root no?

Reply:
I had to re-write this section because when writing tests for this script, temporary directories under `tmp` passed as `coursesPath` would cause `baseDir` to resolve to `/`.

Reply:
Also, looking at the check you had in there:
```
const relative = path.relative(baseDir, zipsPath)
  if (relative && !relative.startsWith("..") && !path.isAbsolute(relative)) {
```
Wouldn't `path.isAbsolute(relative)` always be false?